### ISSUE TITLE
GitBlit not working in iframe

### ISSUE THREAD

**Issue by AndreFabris** [gitblit/gitblit#1319]
Hi,
I am trying to get gitblit to work in iframe. Nothing is displayed
Console shows:
Load denied by X-Frame-Options: http://appsforce.com:8080/ does not permit cross-origin framing.
SecurityError: Permission denied to access property "document" on cross-origin object
Failed to execute ‘postMessage’ on ‘DOMWindow’: The target origin provided (‘http://appsforce.com:8080’) does not match the recipient window’s origin (‘null’).

Iframe is loaded as:
<iframe src="http://' . $x1 . ':8080" name ="appsforceiframe" sandbox="allow-scripts" width="100%" height="1500px" scrolling="no" frameBorder="0"></iframe>
where $x1 = $_SERVER['HTTP_HOST'];

Any ideas? Using latest 1.8.0 version

**Comment by AndreFabris** [comment]
Any update on this one, is anyone even checking these tickets

**Comment by fzs** [comment]
Hi!
Yes, I, for one, read these tickets. 
But to be honest, I have no idea what is going on there, and I have also no existing setup where Gitblit would run in an iframe, no idea what the use case is, or any vested interest in doing that. So  for me, this is unfortunately in the category of "when I have time and nothing is more pressing". 

The hope would be that you are not the only one with that problem, and that someone else in the community of gitblit users might have a solution or may have fixed it. But I am guessing that not too many people are running Gitblit in an iframe.

I'm sorry I can't help much more with that issue right now. :(

**Comment by boughtonp** [comment]
The [X-Frame-Options](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Frame-Options) header is what's stopping it working.

The header is being explicitly set here:
https://github.com/gitblit/gitblit/tree/master/src/main/java/com/gitblit/wicket/pages/BasePage.java#L183

I guess the options are either compile your own version without that line, or make it a configurable preference and submit a pull request.

**Comment by fzs** [comment]
The original issue is #796 which lead to the header being added. But that mentioned that it is a known use case to embed Gitblit in an iFrame and the fix tries not to stop it completely. So as long as it is the same origin it should work, it sounds like. 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by boughtonp** [comment]
Yeah, but unfortunately browsers consider `example.com` and `example.com:8080` to be different origins, and didn't bother to add, e.g. a samedomain option.

If that is the OP's situation then another option might be a reverse proxy to mask the port behind a URL prefix (e.g. proxy `example.com/8080/*` to `example.com:8080/*`)
